src/NuGet.Core/NuGet.Commands/RestoreCommand/RequestFactory/RestoreArgs.cs

```diff
@@ -119,12 +120,18 @@ private List<SourceRepository> GetEffectiveSourcesCore(ISettings settings)
             if (packageSources.Count < 1)
```

Comment:
It overrides, but in some scenarios it's also to filter down to a subset. In which case the user might want to only use a single source and the credentials for it. Not a critical scenario though, those users should just pass ``--config-file`` probably